ngx_http_dav_ext_module.c

```diff
@@ -771,15 +771,15 @@ ngx_http_dav_get_xattr(ngx_http_request_t *r, u_char *path, ngx_http_dav_ext_ent
 {
     ssize_t                   xrc;
     char                      xattr_buf[256];
-    void                      *xattr_ptr = xattr_buf;
+    char                      *xattr_ptr = xattr_buf;
     unsigned int              mode;
     int                       rdev_major, rdev_minor, uid, gid;
 
-    xrc = lgetxattr((const char*)path, XSTAT_ATTR, xattr_buf, sizeof(xattr_buf));
+    xrc = lgetxattr((const char*)path, XSTAT_ATTR, xattr_buf, sizeof(xattr_buf) - 1);
```

Comment:
Please describe the meaning of the 'sizeof(xattr_buf) - 1' 
I think that sizeof(xattr_buf) (pointer on 64bit system) is 8.

Reply:
xattr_buf is not a pointer: it's a buffer:
char                      xattr_buf[256];
so sizeof(xatrr_buf) = sizeof(char)*256 = 256

sizeof(xattr_buf) - 1 is to reverse last symbol for '\0' termination symbol: the whole idea of the fix is that lgetxattr() actually do not terminate string that it wrote to the target buffer: this scanf() later can read extra symbol.

Reply:
ok